autoit_ripper/utils.py

```diff
@@ -15,12 +27,68 @@ def bytes_to_bitstring(data: bytes) -> str:
 
 class BitStream:
     def __init__(self, data: bytes) -> None:
-        self.data = bytes_to_bitstring(data)
+        self._data = bytes_to_bitstring(data)
+        self._offset = 0
 
     def get_bits(self, num: int) -> int:
-        out = int(self.data[:num], 2)
-        self.data = self.data[num:]
-        return out
+        data = self._data[self._offset : self._offset + num]
+        self._offset += num
+        return int(data, 2)
+
+
+class ByteStream:
+    def __init__(self, data: bytes) -> None:
+        self._data = data
+        self._offset = 0
+
+    def skip_bytes(self, num: int) -> None:
+        self._offset += num
+
+    def get_bytes(self, num: Optional[int]) -> bytes:
+        if num is None:
+            num = len(self._data) - self._offset
+
+        data = self._data[self._offset : self._offset + num]
+        self._offset += num
+        return data
+
+    def _int(self, len: int, signed: bool) -> int:
+        return int.from_bytes(
+            bytes=self.get_bytes(len), byteorder="little", signed=signed
+        )
+
+    def _int_be(self, len: int, signed: bool) -> int:
+        return int.from_bytes(bytes=self.get_bytes(len), byteorder="big", signed=signed)
+
+    def f64(self) -> float:
+        return unpack_from("<d", self.get_bytes(8))[0]
+
+    def u8(self) -> int:
+        return self._int(1, False)
+
+    def i8(self) -> int:
+        return self._int(1, False)
```

Comment:
Here and below - why even have `signed` bit if it's always false :thinking: Is this intentional?